inductiva/coastal/README.md

```diff
@@ -0,0 +1,93 @@
+# Coastal dynamics
+
+Inductiva API supports simuulation of coastal dynamics via two simulators:
+[SWASH](https://swash.sourceforge.io/) and [XBeach](https://oss.deltares.nl/web/xbeach/).
+These simulators solve the [shallow-water equations](https://en.wikipedia.org/wiki/Shallow_water_equations),
+a reduced version of the Navier-Stokes equations valid for small depth-to-length
+ratios (i.e., shallow water).
+
+## Coastal area scenario
+
+This scenario simulates waves propagating in a coastal area represented by a
+bathymetric profile (i.e., the map of depths of the sea bottom as a function
+of spatial coordinates x, y). Waves are injected into the domain from one of the
+boundaries of the simulation with a given amplitude and period. The simulation
+is performed using the [SWASH](https://swash.sourceforge.io/) simulator.
+
+### Example
+
+Start by creating or loading a bathymetry. This can be done e.g. by reading an
+[ASCII XYZ file](https://emodnet.ec.europa.eu/sites/emodnet.ec.europa.eu/files/public/20171127_DTM_exchange_format_specification_v1.6.pdf),
+a standard format for bathymetry data (for examples, see e.g. the [European Marine Observation and Data Network](https://emodnet.ec.europa.eu/geoviewer/#!/)).
+Here we shall use the bathymetry data from Algarve available [here](https://sextant.ifremer.fr/record/SDN_CPRD_590_HR_Lidar_Algarve/)
+as an example.
+
+```python
+import inductiva
+
+bathymetry = inductiva.coastal.Bathymety.from_ascii_xyz_file(
+    ascii_xyz_file_path="590_HR_Lidar_Algarve.emo")
+```
+
+The bathymetry can be inspected by plotting it with a given resolution, in this
+case 200 m x 200 m:
+
+```python
+bathymetry.plot(x_resolution=200, y_resolution=200)
+```
+
+![Raw bathymetry.](resources/media/bathymetry.png)
+
+Raw bathymetry data will often span an area too large and sparsely sampled to
+be used in a simulation. In this case, the user can crop the bathymetry to a
+smaller area of interest:
+
+```python
+bathymetry = bathymetry.crop(x_range=(51000, 52000), y_range=(12150, 13000))
+bathymetry.plot()
+```
+
+![Cropped bathymetry.](resources/media/bathymetry_cropped.png)
+
+To be used in a simulation, the bathymetry data must be interpolated to a
+regular grid with custom resolution, in this case 5 m x 5 m. In case some
+positions are missing in the bathymetry data (see e.g. the lower left corner in
+the figure above), the user can choose to fill them with a constant depth or the
+nearest depth value:
+
+```python
+bathymetry = bathymetry.to_uniform_grid(x_resolution=5,
```

Comment:
Shall we say what happens if people do not do it? Don't we try by default to do it?